### ISSUE TITLE
Support for custom views.

### ISSUE THREAD

**Issue by AOrobator** [soarcn/BottomSheet#43]
There should be support for custom views inside the bottom sheet.

**Comment by soarcn** [comment]
Yes, I will support this, and first thing gonna happen is to support action items customization.

**Comment by mkaarthick** [comment]
Is there support for sheet coming from top?

**Comment by soarcn** [comment]
@mkaarthick I don't have plan to support sheet coming from top, but you can try it in your fork, check #37 for more info

**Comment by jjhesk** [comment]
you can using sliding panel for this.

**Comment by mkaarthick** [comment]
@jjhesk Sliding Panel by default comes from left to right. How can I make it from top to bottom?

**Comment by soarcn** [comment]
I try to understand why put dialog to the top part of screen.

**Comment by mkaarthick** [comment]
@soarcn I didn't but client need it in that way.

**Comment by redmanit** [comment]
So when will BottomSheet support custom view?

**Comment by livotov** [comment]
We've implemented the custom view support in our clone of this project as we needed it urgently. If @soarcn have no objectives, I can port it here and create a pull request with the changes.

Basically, we've implemented the custom view supoprt the following way:
```
builder....customView(myView).customViewResultListener(customViewResultListener).show();
```

when the view is ready to be closed, it it's own responsibility to call:
```
bottomSheetInstance.onCustomViewFinished(customResultObject)
```

If custom view class implements the BottomSheetCustomViewAdapter interface, the instance of BottomSheet will be automatically injected via the setHost method. Otherwise custom view have to obtain it's BottomSheet class instance on its own.

**Comment by soarcn** [comment]
developers can customize the whole ui by themself from next release (different from @livotov 's  way,  you can only change the ui for current stage)
I will update readme on how to customize the ui later,  short summary here.
```xml
        <attr name="bs_listItemLayout" format="reference" />
        <attr name="bs_gridItemLayout" format="reference" />
        <attr name="bs_headerLayout" format="reference" />
````
you need to define your own layout with given view ids, just like bs_list_entry.xml

@livotov Love to view your clone, can you show me the link please?

**Comment by livotov** [comment]
@soarcn but doesn't this mean (your way) that custom ui will still need  to operate with the list view ?
To my mind, your way (complete sheet UI customization) will be a great companion to mine one, forming the complete customization solution. Because sometimes one need just to use your own (non list-based) view in one or two sheets while use standard menu items in others. And in parallel one may want to tweak the bottom sheet appearance by tweaking layout files. So would be nice to have both possibilities, I think.

Regarding the clone - we're currently working on V3 of our internal helper library https://github.com/LivotovLabs/RoboTools, so you can see the modified code there. The codebase is dirty at the moment and will change a lot soon as we're in progress refactoring it, but you can see the idea there.

**Comment by soarcn** [comment]
@livotov   Thanks for you comments.

I don't have plan (currently) to support fully customization now, because 
1, The api would be quite complicated.
2, If you need a totally different looking bottomsheet, you might using the wrong UI component.
3, Bottomsheet use menuitem as its model, supporting more attributes(to display) that menuitem have is not making sence.
